oshmem/mca/sshmem/sysv/sshmem_sysv_component.c

```
@@ -111,15 +111,30 @@ sysv_runtime_query(mca_base_module_t **module,
     /* if we are here, then let the run-time test games begin */
 
 #if defined (SHM_HUGETLB)
-    mca_sshmem_sysv_component.use_hp = 1;
-    flags = IPC_CREAT | IPC_EXCL | S_IRUSR | S_IWUSR | SHM_HUGETLB;
-    if (-1 == (shmid = shmget(IPC_PRIVATE, (size_t)(opal_getpagesize()), flags))) {
-        mca_sshmem_sysv_component.use_hp = 0;
+    if (mca_sshmem_sysv_component.use_hp > 0) {
+         flags = IPC_CREAT | IPC_EXCL | S_IRUSR | S_IWUSR | SHM_HUGETLB;
+        if (-1 == (shmid = shmget(IPC_PRIVATE, oshmem_gethugepagesize(), flags))) {
+            if (mca_sshmem_sysv_component.use_hp == 1) {
+                mca_sshmem_sysv_component.use_hp = 0;
+                goto out;
+            }
+            mca_sshmem_sysv_component.use_hp = 0;
+        }
+        else if ((void *)-1 == (addr = shmat(shmid, NULL, 0))) {
+            shmctl(shmid, IPC_RMID, NULL);
+            if (mca_sshmem_sysv_component.use_hp == 1) {
+                mca_sshmem_sysv_component.use_hp = 0;
+                goto out;
+            }
+            mca_sshmem_sysv_component.use_hp = 0;
+        }
     }
-    else if ((void *)-1 == (addr = shmat(shmid, NULL, 0))) {
-        shmctl(shmid, IPC_RMID, NULL );
+#else
+    if (mca_sshmem_sysv_component.use_hp == 1) {
         mca_sshmem_sysv_component.use_hp = 0;
+        goto out;
```

Shouldn't it be an error if the user asked for this functionality (i.e., use_hp==1) but you can't deliver it?  It should only be silently ignored if use_hp indicates "automatic" use.

Right?